### DISCUSSION TITLE
Can a feature like echo cancelling in parsec be added?

### DISCUSSION THREAD

**Discussion by WangChuDi** [ClassicOldSong/Apollo#52] [Ideas]
I often use apollo to play games on one computer with friends and use voice software. However, Sunshine/Apollo streaming will capture the sound of the voice software. Can a feature like echo cancelling in parsec be added?

![image](https://github.com/user-attachments/assets/b216cb6a-6c6a-41d1-9452-db961edf2a1a)

**Comment by ClassicOldSong** [comment]
If you want to exclude voice from discord, there's currently no method for this from the computer. You can use a phone or other device for voice calls for now as a temporary solution.

**Comment by ClassicOldSong** [comment]
You can try tinkering with https://vb-audio.com/Voicemeeter/index.htm to see if it solves your problem

Basically, set the audion device to be captured to the virtual audio interface created by Voicemeter, and set the discord output to your physical output. Also route the virtual audio output to your physical output through Voicemeter.

**Comment by WangChuDi** [comment]
> You can try tinkering with https://vb-audio.com/Voicemeeter/index.htm to see if it solves your problem
> 
> Basically, set the audion device to be captured to the virtual audio interface created by Voicemeter, and set the discord output to your physical output. Also route the virtual audio output to your physical output through Voicemeter.

Thank you for your solution! This does achieve this feature.

**Comment by WangChuDi** [comment]
Found some description  in Parsec: https://support.parsec.app/hc/en-us/articles/115003042152-Eliminating-Sound-Echo-While-Co-Op-ing-With-Friends

> Due to how this option works, Parsec captures audio from all audio devices on your system at once, like your headphones, speakers and virtual outputs all at the same time.

Maybe it will help you in your work.

**Comment by ClassicOldSong** [comment]
The method mentioned in Parsec is basically the same as I provided prioror. Audio routing issue will be handled in the future, but it's not a priority now.
